.github/workflows/charmcraft-pack.yaml

```diff
@@ -0,0 +1,38 @@
+name: Charmcraft Pack Test
+
+on: [push, pull_request, workflow_call]
+
+jobs:
+  charmcraft-pack:
+    runs-on: ubuntu-20.04
+
+    steps:
+    - name: Set up Python 3.8
+      uses: actions/setup-python@v2
+      with:
+        python-version: "3.8"
+
+    - name: Checkout test charm repository
+      uses: actions/checkout@v2
+      with:
+        repository: jnsgruk/hello-kubecon
+
+    - name: Update 'ops' dependency in test charm to latest
+      run: |
+        # We need to reference the PR branch's repo and commit (not the GITHUB_SHA
+        # temporary merge commit), because charmcraft pack does a git checkout which
+        # can't see the temporary merge commit.
+        sed -i -e "/^ops[ ><=]/d" -e "/canonical\/operator/d" -e "/#egg=ops/d" requirements.txt
+        echo -e "\ngit+${{ github.event.pull_request.head.repo.clone_url }}@${{ github.event.pull_request.head.sha }}#egg=ops" >> requirements.txt
```

Comment:
We use the same sed + echo incantation in our other CI jobs to replace the "ops" requirement with "git+https://github.com/canonical/operator@$GITHUB_SHA#egg=ops" so we're using the PR version of ops. It's the same here, except that we need to do it against the PR branch itself, not the temp merge commit (as per the comment).

If it fails -- and I had a number of failures getting this right :-) -- we'll almost certainly get an error when we try to `charmcraft pack`. For example: incorrect syntax, wrong hash, etc (I got all those errors when testing). That's part of the reason I'm doing `cat requirements.txt` below so it's clear exactly what we're trying to install.

During testing, I also made sure it failed first with the failure I was expecting, then I merged the setup.py fix from main, and then it passed. So I'm confident this is right.

Reply:
Heh, though of course it broke once it was merged into main, because there `${{ github.event.pull_request.head.sha }}` isn't set. :-) Fixing in https://github.com/canonical/operator/pull/916